packages/inject/src/index.js

```diff
@@ -167,7 +167,7 @@ export default function inject(options) {
 
           // special case – shorthand properties. because node.key === node.value,
           // we can't differentiate once we've descended into the node
-          if (node.type === 'Property' && node.shorthand) {
+          if (node.type === 'Property' && node.shorthand && node.value.type === 'Identifier') {
```

Comment:
My worry is that with this change, if the shorthand property is an assignment pattern but the **left** side should be replaced, it would cause the same issue we are guarding against.

i.e. what would happen here

```js
export const test = ({process = {}}) => {
   console.log(process);
 };
```

(Admittedly I did not run the code...)
Maybe we should rather add special handling for assignment patterns in shorthands than restrict the current special handling to the identifier case?

Reply:
I think it should be guarded by `if (scope.contains(...))` since `process` is a local variable in this case.

I guess we can add more test cases e.g.
```js
{
  const {$} = foo;
}
{
  const foo = {$};
}
{
  const foo = ({$}) => {};
}
{
  const foo = ({bar = $}) => {};
}

```